refactor: replace react-toastify with NotificationToast in UserPortal components (Phase 3B) #5203

MuhammadUmar7195 opened the pull request:
### What kind of change does this PR introduce?

Refactoring - Replacing direct `react-toastify` calls with the standardized `NotificationToast` component in UserPortal components.

### **Issue Number:**

<!--Add related issue number here.-->
Fixes #5203 

### **Snapshots/Videos:**
N/A - This is a refactoring change with no visual changes. The toast notifications will appear the same to users.

### **If relevant, did you update the documentation?**
N/A - No documentation changes required as this is an internal refactoring.

### **Summary**
This PR refactors high-priority UserPortal components and their corresponding test files to use the standardized `NotificationToast` component instead of `react-toastify`. This ensures UI consistency and better error-handling patterns across the portal.

### **Changes Made**
Based on the modified files in the source control, the following refactors were completed:

**Component Refactors**
Replaced `toast.success()`, `toast.error()`, and `toast.warn()` with NotificationToast methods in:
- `EventCard.tsx`
- `Donate.tsx`
- `Events.tsx`
- `LeaveOrganization.tsx`
- `Settings.tsx`
- `GroupModal.tsx`
- `Invitations.tsx`
- `UpcomingEvents.tsx`

### **Does this PR introduce a breaking change?**
No - This is a refactoring change that maintains the same user experience. All toast notifications will function identically to before, just using the new standardized component.

## Checklist

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [x] I have implemented or provided justification for each non-critical suggestion
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added search in volunteer and donate areas; improved volunteer success messaging.

* **Improvements**
  * Unified notification UX across user flows (success/error messages standardized).
  * Better profile update and file-upload validation messages (password length, file type/size, image processing).
  * Minor UI icon and spacing adjustments.

* **Tests**
  * Updated test suites to align with the unified notification behavior and messaging.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes commented:
Please refer to this:
- https://docs-admin.talawa.io/docs/developer-resources/testing#pre-commit-hooks

MuhammadUmar7195 commented:
@palisadoes, I've refactored the UserPortal components to `NotificationToast`, addressed all CodeRabbit suggestions, and ensured all test cases pass. Could you please review my PR when you're free?

MuhammadUmar7195 commented:
@palisadoes All CodeRabbit feedback has been addressed:
- CSS policy violations fixed (removed inline styles, camelCase props)
- i18n violations fixed (all error messages now use translation keys)
- French locale uses proper notation (5 Mo instead of 5MB)

**Now its ready for review**

MuhammadUmar7195 commented:
@palisadoes I have fixed the Shard 3 assertion failures by aligning the `en/translation.json` strings with the `Settings.spec.tsx` test expectations. Local verification confirms all 18 Settings tests pass, though an unrelated failure remains in `OrganizationPeople.spec.tsx` Can you please help me out for this issue to clear my effort 

<img width="1094" height="617" alt="image" src="https://github.com/user-attachments/assets/ed515a37-d182-44d0-8e57-ff8afa91c970" />

MuhammadUmar7195 commented on src/screens/UserPortal/Volunteer/UpcomingEvents/UpcomingEvents.tsx:
```diff
@@ -424,7 +424,7 @@ const UpcomingEvents = (): JSX.Element => {
       <div className={`${styles.container} bg-white rounded-4 my-3`}>
         <div className={styles.message} data-testid="errorMsg">
-          <WarningAmberRounded className={styles.errorIcon} fontSize="large" />
+          <WarningAmberRounded className={styles.errorIcon} />
```
Without this i face an CSS policy inline issue that's why i resolve it

MuhammadUmar7195 commented:
@palisadoes  I've verified all translation keys `(volunteer, volunteered, volunteerGroups, volunteerRequestSuccess)` exist in all 5 locale files `en, es, fr, hi, zh`. The keys are present at lines `531-536` and `1764`. All 18 Settings tests pass locally. This might be a CI caching issue. Could you please re-run the workflow?

<img width="900" height="600" alt="image" src="https://github.com/user-attachments/assets/d8be35f8-24d4-4934-ad1b-143e3779e19a" />

palisadoes commented:
Please add them. The files you edited don't have the proper translations applied. It's not just related to the lines you edited within the files

MuhammadUmar7195 commented:
@palisadoes Resolve all conflicts, Please review it

palisadoes commented:
@coderabbitai full review

palisadoes commented:
FYI

This issue is blocking other important issues. PTAL, and if interested use the auto-assign feature to tackle the challenge

- https://github.com/PalisadoesFoundation/talawa-admin/issues/5085
